embedded-graphics/src/primitives/polyline.rs

```diff
@@ -0,0 +1,426 @@
+    /// assert_eq!(moved.bounding_box().top_left, Point::new(15, 19));
+    /// ```
+    fn translate(&self, by: Point) -> Self {
+        Self {
+            translate: by,
+            ..*self
+        }
+    }
+
+    /// Translate the polyline from its current position to a new position by (x, y) pixels.
+    ///
+    /// ```
+    /// # use embedded_graphics::primitives::Polyline;
+    /// # use embedded_graphics::prelude::*;
+    /// let points = [
+    ///     Point::new(5, 10),
+    ///     Point::new(7, 7),
+    ///     Point::new(5, 8),
+    ///     Point::new(10, 10),
+    /// ];
+    ///
+    /// let mut polyline = Polyline::new(&points);
+    ///
+    /// polyline.translate_mut(Point::new(10, 12));
+    ///
+    /// assert_eq!(polyline.bounding_box().top_left, Point::new(15, 19));
+    /// ```
+    fn translate_mut(&mut self, by: Point) -> &mut Self {
+        self.translate += by;
+
+        self
+    }
+}
+
+/// An iterator over all pixel positions on the polyline
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct Points<'a> {
+    vertices: &'a [Point],
+    translate: Point,
+    segment_iter: line::Points,
+}
+
+impl<'a> Points<'a> {
+    fn new<'b>(polyline: &'b Polyline<'a>) -> Self
+    where
+        'a: 'b,
+    {
+        polyline
+            .vertices
+            .split_first()
+            .and_then(|(start, rest)| {
+                // Polyline is 2 or more vertices long, return an iterator for it
+                rest.get(0).map(|end| Points {
+                    vertices: rest,
+                    translate: polyline.translate,
+                    segment_iter: Line::new(*start + polyline.translate, *end + polyline.translate)
+                        .points(),
+                })
+            })
+            .unwrap_or_else(||
+                // Polyline is less than 2 vertices long. Return a dummy iterator that will short
+                // circuit 
+                Points {
+                    vertices: &[],
+                    translate: Point::zero(),
+                    segment_iter: Line::new(Point::zero(), Point::zero()).points()
+                })
+    }
+}
+
+impl<'a> Iterator for Points<'a> {
+    type Item = Point;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if let Some(p) = self.segment_iter.next() {
+            Some(p)
+        } else {
+            let (start, rest) = self.vertices.split_first()?;
+            let end = rest.get(0)?;
+
+            self.vertices = rest;
+
+            self.segment_iter = Line::new(*start + self.translate, *end + self.translate).points();
+
+            // Skip first point of next line, otherwise we overlap with the previous line
+            self.nth(1)
+        }
+    }
+}
+
+impl<'a, C> IntoIterator for &'a Styled<Polyline<'a>, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+    type IntoIter = StyledPolylineIterator<'a, C>;
+
+    fn into_iter(self) -> Self::IntoIter {
```

Comment:
I started to move the code from the `into_iter` methods to `StyledPolylineIterator::new`. This helps if we decide to split the primitives files if they get too long by keeping related code together in one place.